planner/core/testdata/integration_serial_suite_out.json

```diff
@@ -113,6 +113,26 @@
           "└─TableFullScan_10 10000.00 cop[tiflash] table:ttt keep order:true, stats:pseudo"
         ],
         "Warn": null
+      },
+      {
+        "SQL": "desc select /*+ read_from_storage(tikv[t]) */ * from ttt",
+        "Plan": [
+          "TableReader_7 10000.00 root  data:TableFullScan_6",
+          "└─TableFullScan_6 10000.00 cop[tiflash] table:ttt keep order:false, stats:pseudo"
+        ],
+        "Warn": [
+          "[planner:1815]There are no matching table names for (t) in optimizer hint /*+ READ_FROM_STORAGE(t) */. Maybe you can use the table alias name"
+        ]
+      },
+      {
+        "SQL": "desc select /*+ read_from_storage(tiflash[t], tikv[tt]) */ * from ttt",
+        "Plan": [
+          "TableReader_7 10000.00 root  data:TableFullScan_6",
+          "└─TableFullScan_6 10000.00 cop[tiflash] table:ttt keep order:false, stats:pseudo"
+        ],
+        "Warn": [
+          "[planner:1815]There are no matching table names for (t, tt) in optimizer hint /*+ READ_FROM_STORAGE(t, tt) */. Maybe you can use the table alias name"
```

Comment:
- The hint user inputs is `/*+ read_from_storage(tiflash[t], tikv[tt]) */`
- The hint TiDb outputs is `/*+ READ_FROM_STORAGE(t, tt) */`

There are two problems:

1. the output is mismatched with the input, which makes user confused.
2. the output hint syntax is invalid.


`appendUnmatchedJoinHintWarning()` is used for join related hints, it's not suitable to be used for `read_from_storage`.